xcode/Subconscious/Shared/Services/NoosphereService.swift

```diff
@@ -18,6 +18,99 @@ import SwiftNoosphere
 public enum NoosphereError: Error {
     /// Thrown when something unexpected happens on the other side of the FFI, and we don't know what went wrong.
     case foreignError(String)
+    case nullPointer
+}
+
+struct NoosphereFFI {
+    static func callWithError<Z>(
```

Comment:
Error handling helper. Executes ffi function in a closure, checks for errors, handles the various cases, and throws a Swift error if anything went wrong.

Test in https://github.com/subconsciousnetwork/subconscious/pull/396/files#diff-7b15ecfa4b38a8a713d86708d79b17145c1c2ae684d52fba34f8f5c0035311bb